configure.sh

```diff
@@ -1,29 +1,75 @@
 #!/bin/bash
-if ! type parity > /dev/null; 
-then
+
+# Install parity if this is a linux system
+#
+installLinux() {
     PARITY_DEB_URL=https://parity-downloads-mirror.parity.io/v1.8.6/x86_64-unknown-linux-gnu/parity_1.8.6_amd64.deb
-    if ! type curl > /dev/null; 
+    if ! type curl > /dev/null;
     then
         echo "Installing curl."
         sudo apt-get update -qq
         sudo apt-get install -y -qq curl
-    else 
+    else
         echo "Curl is already installed."
     fi
     echo "Installing parity"
     file=/tmp/parity.deb
     curl -Lk $PARITY_DEB_URL > $file
     sudo dpkg -i $file
     rm $file
-else 
+}
+
+# Install parity if this is a Darwin system (macOS)
+#
+installMac() {
+
+    if  !type brew > /dev/null;
+    then
+
+        echo "This requires Brew."
+        echo "Installing Homebew"
+        /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+    fi
+
+    echo "Installing parity"
+
+    brew tap paritytech/paritytech
+    brew install parity
+}
+
+
+if ! type parity > /dev/null;
+then
+
+    platform='unknown'
+    unamestr=`uname`
+    if [[ "$unamestr" == 'Linux' ]]; then
+
+       installLinux;
+    elif [[ "$unamestr" == 'Darwin' ]]; then
+       installMac;
+    else
+        echo "Platform: $unamestr unknown";
+        exit -1;
+    fi
+
+
+else
     echo "Parity is already installed."
 fi
 
 
+
+
```

Comment:
Random whitespace at the bottom here probably doesn't have to be in here. There are also newlines in the functions that I find make it a little trickier to read (e.g. 56-57, 48).